lib/src/trailer.rs

```diff
@@ -0,0 +1,264 @@
+    fn test_trailers_with_colon_in_body() {
+        let body = r#"chore: update itertools to version 0.14.0
+
+Summary: Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
+tempor incididunt ut labore et dolore magna aliqua.
+
+Change-Id: I1234567890abcdef1234567890abcdef12345678"#;
+
+        let trailer = parse_description_trailers(body);
+
+        // should only have Change-Id
+        assert_eq!(trailer.len(), 1);
+        assert_eq!(trailer[0].key, "Change-Id");
+    }
+
+    #[test]
+    fn test_multiline_trailer() {
+        let body = r#"chore: update itertools to version 0.14.0
+
+key: This is a very long value, with spaces and
+  newlines in it."#;
+
+        let trailer = parse_description_trailers(body);
+
+        // should only have Change-Id
+        assert_eq!(trailer.len(), 1);
+        assert_eq!(trailer[0].key, "key");
+        assert!(trailer[0].value.starts_with("This is"));
+        assert!(trailer[0].value.ends_with("in it."));
+    }
+
+    #[test]
+    fn test_ignore_line_in_trailer() {
+        let body = r#"chore: update itertools to version 0.14.0
+
+Signed-off-by: Random J Developer <random@developer.example.org>
+[lucky@maintainer.example.org: struct foo moved from foo.c to foo.h]
+Signed-off-by: Lucky K Maintainer <lucky@maintainer.example.org>
+"#;
+
+        let trailer = parse_description_trailers(body);
+        assert_eq!(trailer.len(), 2);
+    }
+
+    #[test]
+    fn test_trailers_with_single_line_description() {
+        let body = r#"chore: update itertools to version 0.14.0"#;
+        let trailer = parse_description_trailers(body);
+        assert_eq!(trailer.len(), 0);
+    }
+
+    #[test]
+    fn test_blank_line_after_trailer() {
+        let body = r#"subject
+
+foo: 1
+
+"#;
+        let trailer = parse_description_trailers(body);
+        assert_eq!(trailer.len(), 1);
+    }
+
+    #[test]
+    fn test_blank_line_inbetween() {
+        let body = r#"subject
+
+foo: 1
+
+bar: 2
+"#;
+        let trailer = parse_description_trailers(body);
+        assert_eq!(trailer.len(), 1);
+    }
+
+    #[test]
+    fn test_no_blank_line() {
+        let body = r#"subject: whatever
+foo: 1
+"#;
+        let trailer = parse_description_trailers(body);
+        assert_eq!(trailer.len(), 0);
+    }
+
+    #[test]
+    fn test_whitespace_before_key() {
+        let body = r#"subject
+
+ foo: 1
+"#;
+        let trailer = parse_description_trailers(body);
+        assert_eq!(trailer.len(), 0);
+    }
+
+    #[test]
+    fn test_whitespace_after_key() {
+        let body = r#"subject
+
+foo : 1
+"#;
+        let trailer = parse_description_trailers(body);
+        assert_eq!(trailer.len(), 1);
+        assert_eq!(trailer[0].key, "foo");
+    }
+
+    #[test]
+    fn test_whitespace_around_value() {
+        let body = r#"subject
+
+foo :  1 
+"#;
+        let trailer = parse_description_trailers(body);
+        assert_eq!(trailer.len(), 1);
+        assert_eq!(trailer[0].value, "1");
+    }
```

Comment:
I guess we should have one without the space before the value. I.e. something like this:

```
subject

foo:1
```